ipld/routing.ipldsch

```diff
@@ -4,20 +4,39 @@ type Envelope union {
      | GetP2PProvideResponse "get-p2p-provide-response"
 } representation keyed
```

Comment:
@warpfork continuing from https://github.com/ipfs/go-delegated-routing/pull/7#issuecomment-987012120 (GitHub's lack of threads are brutal and I'd like to isolate this topic a bit).

> hen there's not much useful that can be done other than describing the data as Any, as far as I can figure.

Can `Any` currently be parsed by codegen?

> but that's an accurate representation of what would be mechanically needed, even if there were syntactic facades over it; I'm not actually sure we want such a thing to be easy to describe: it's sort of a negative-feedback form of mechanical sympathy

Is this necessarily true? Say I have a keyed union for the keys "foo" and "bar", but my parser sees a "baz". Why would having a keyed union like the below be so rough to process, as soon as we notice the key isn't "foo" or "bar" we just treat it like an `Any` which offhand doesn't seem like it'd be more work than processing a regular field that happens to be an `Any`.

```
type OpenKeyedUnion union {
     | String "foo"
     | String "baz"
     | Any
} representation keyed
```

Also, some union types are more efficient than others for parsing so adding one more that eases consumer pain here doesn't seem too bad, but I could be missing things so lmk 😄.